cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -80,7 +80,8 @@ public FactTable get(int resolution) {
     int ttl =  cConf.getInt(MetricsConstants.ConfigKeys.RETENTION_SECONDS + "." + resolution + ".seconds", -1);
 
     DatasetProperties.Builder props = DatasetProperties.builder();
-    if (ttl > 0) {
+    // don't add TTL for MAX_RESOLUTION table.
+    if (ttl > 0 && resolution != Integer.MAX_VALUE) {
```

Comment:
alright - let's override it for now. Can you add in that comment mention about jira issue "CDAP-1626" please.